app/assets/javascripts/dashboard/views/dash/explorative_annotations_view.js

```diff
@@ -0,0 +1,354 @@
+    unarchivedTracings: Array<APIAnnotationType>,
+    alreadyFetchedMetaInfo: {
+      archived: boolean,
+      unarchived: boolean,
+    },
+    searchQuery: string,
+    requestCount: number,
+  } = {
+    showArchivedTracings: false,
+    archivedTracings: [],
+    unarchivedTracings: [],
+    alreadyFetchedMetaInfo: {
+      archived: false,
+      unarchived: false,
+    },
+    searchQuery: "",
+    requestCount: 0,
+  };
+
+  componentDidMount() {
+    this.fetchDataMaybe();
+  }
+
+  isFetchNecessary(): boolean {
+    const accessor = this.state.showArchivedTracings ? "archived" : "unarchived";
+    return !this.state.alreadyFetchedMetaInfo[accessor];
+  }
+
+  enterRequest() {
+    this.setState({ requestCount: this.state.requestCount + 1 });
+  }
+
+  leaveRequest() {
+    this.setState({ requestCount: this.state.requestCount - 1 });
+  }
+
+  async fetchDataMaybe(): Promise<void> {
+    if (!this.isFetchNecessary()) {
+      return;
+    }
+
+    const isFinishedString = this.state.showArchivedTracings.toString();
+    const url = this.props.userID
+      ? `/api/users/${this.props.userID}/annotations?isFinished=${isFinishedString}`
+      : `/api/user/annotations?isFinished=${isFinishedString}`;
+
+    this.enterRequest();
+    const tracings = await Request.receiveJSON(url);
+    this.leaveRequest();
+    if (this.state.showArchivedTracings) {
+      this.setState(
+        update(this.state, {
+          archivedTracings: { $set: tracings },
+          alreadyFetchedMetaInfo: {
+            archived: { $set: true },
+          },
+        }),
+      );
+    } else {
+      this.setState(
+        update(this.state, {
+          unarchivedTracings: { $set: tracings },
+          alreadyFetchedMetaInfo: {
+            unarchived: { $set: true },
+          },
+        }),
+      );
+    }
+  }
+
+  toggleShowArchived = () => {
+    this.setState({ showArchivedTracings: !this.state.showArchivedTracings }, () =>
+      this.fetchDataMaybe(),
+    );
+  };
+
+  finishOrReopenTracing = (type: "finish" | "reopen", tracing: APIAnnotationType) => {
+    const controller = jsRoutes.controllers.AnnotationController;
+    const url =
+      type === "finish"
+        ? controller.finish(tracing.typ, tracing.id).url
+        : controller.reopen(tracing.typ, tracing.id).url;
+
+    Request.receiveJSON(url).then(newTracing => {
+      Toast.message(newTracing.messages);
+
+      const newTracings = this.getCurrentTracings().filter(t => t.id !== tracing.id);
+
+      if (type === "finish") {
+        this.setState({
+          unarchivedTracings: newTracings,
+          archivedTracings: [newTracing].concat(this.state.archivedTracings),
+        });
+      } else {
+        this.setState({
+          archivedTracings: newTracings,
+          unarchivedTracings: [newTracing].concat(this.state.unarchivedTracings),
+        });
+      }
+    });
+  };
+
+  renderActions = (tracing: APIAnnotationType) => {
+    if (tracing.typ !== "Explorational") {
```

Comment:
I'm not too sure. The server should only send explorationals, but this check existed in the backbone code. Do you think we should remove it?